includes/TripalFields/ncit__raw_data/ncit__raw_data.inc

```diff
@@ -0,0 +1,276 @@
+      $icons['download'] = $icon_path . 'icon-download.jpg';
+      
+      $traits = [];
+      // This query is identical to the rawphenotypes download page.
+      // Get all experiments (by plant id) where germplasm was used.
+      $all_experiment_locations = chado_query("
+        SELECT p2.project_id, p2.name, value AS location
+        FROM pheno_plantprop 
+          INNER JOIN pheno_plant_project AS p1 USING (plant_id) 
+          INNER JOIN {project} AS p2 ON p1.project_id = p2.project_id
+        WHERE
+          type_id = (SELECT cvterm_id FROM {cvterm} cvt LEFT JOIN {cv} cv ON cv.cv_id = cvt.cv_id
+            WHERE cvt.name = 'Location' AND cv.name = 'phenotype_plant_property_types') AND
+          plant_id IN (SELECT plant_id FROM pheno_plant WHERE stock_id = :germplasm GROUP BY plant_id)
+        GROUP BY p2.project_id, p2.name, value
+      ", array(':germplasm' => $germplasm['id']));
+      $experiment_locations = $all_experiment_locations->fetchAll();
+
+      // All traits in experiment and location.
+      $sql_cvterm = "
+        SELECT c_j.cvterm_json->>'id', c_j.cvterm_json->>'name' FROM (
+          SELECT JSON_BUILD_OBJECT('id', cvterm_id, 'name', name) AS cvterm_json FROM {cvterm} WHERE cvterm_id = ANY ((
+            SELECT STRING_TO_ARRAY(list_id.all_traits, ',') FROM (
+              SELECT string_agg(DISTINCT all_traits, ',') AS all_traits
+              FROM {rawpheno_rawdata_mview}
+              WHERE
+                location IN(:location)
+                AND plant_id IN (SELECT plant_id FROM pheno_plant_project WHERE project_id = :project_id)
+            ) AS list_id
+          )::int[])
+        ) AS c_j
+        WHERE c_j.cvterm_json->>'name' NOT IN ('Rep', 'Entry', 'Location', 'Name', 'Plot', 'Planting Date (date)', '# of Seeds Planted (count)')
+        ORDER BY c_j.cvterm_json->>'name' ASC
+      ";
+      
+      $trait_experiment_location = array();
+      $cache_exp = array();
+      $cache_loc = array();
+      foreach($experiment_locations as $item) {
+        $cache_exp[] = $item->project_id;
+        $cache_loc[] = $item->location;
+
+        $trait_set = chado_query($sql_cvterm, array(':location' => $item->location, ':project_id' => $item->project_id))
+          ->fetchAllKeyed(0, 1);
+        
+        if ($trait_set) {
+          foreach($trait_set as $trait_id => $trait_name) {
+            // Trait id, project id and name + location:
+            $trait_experiment_location[ $trait_id . '_' . $trait_name ][] = $trait_id . '#' . $item->project_id . '#' . $item->name . '#' . $item->location;            
+          }
+        }
+      }
+
+      ksort($trait_experiment_location);
+
+      // Summarize.
+      $summary['traits'] = count($trait_experiment_location);
+      $summary['experiments'] = count(array_unique($cache_exp));
+      $summary['locations'] = count(array_unique($cache_loc));
+
+      $entity->{$field_name}['und'][0]['value']['NCIT:Raw Data'] = array(
```

Comment:
I don't think the value should not be keyed by the field name. Also, you should always use the new array notation: `[ ]` rather than the old style for PHP compatibility reasons.

```suggestion
      $entity->{$field_name}['und'][0]['value'] = array(
```

Reply:
All of the keys exposed through the value should be cvterms and should only represent data. The icons feel out of place here as it relates to the format, not the data. All sub-keys as well so the counts in the summary should also have cvterms as keys.